packages/ledgerstate/color.go

```diff
@@ -0,0 +1,228 @@
+var ColorMint = Color{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region ColoredBalances //////////////////////////////////////////////////////////////////////////////////////////////
+
+// ColoredBalances represents a collection of balances associated to their respective Color that maintains a
+// deterministic order of the present Colors.
+type ColoredBalances struct {
+	balances *orderedmap.OrderedMap
+}
+
+// NewColoredBalances returns a new deterministically ordered collection of ColoredBalances.
+func NewColoredBalances(balances map[Color]uint64) (coloredBalances *ColoredBalances) {
+	coloredBalances = &ColoredBalances{balances: orderedmap.New()}
+
+	// deterministically sort colors
+	sortedColors := make([]Color, 0, len(balances))
+	for color := range balances {
+		sortedColors = append(sortedColors, color)
+	}
+	sort.Slice(sortedColors, func(i, j int) bool { return bytes.Compare(sortedColors[i][:], sortedColors[j][:]) < 0 })
+
+	// add sorted colors to the underlying map
+	for _, color := range sortedColors {
+		coloredBalances.balances.Set(color, balances[color])
+	}
+
+	return
+}
+
+// ColoredBalancesFromBytes unmarshals ColoredBalances from a sequence of bytes.
+func ColoredBalancesFromBytes(bytes []byte) (coloredBalances *ColoredBalances, consumedBytes int, err error) {
+	marshalUtil := marshalutil.New(bytes)
+	if coloredBalances, err = ColoredBalancesFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse ColoredBalances from MarshalUtil: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ColoredBalancesFromMarshalUtil unmarshals ColoredBalances using a MarshalUtil (for easier unmarshaling).
+func ColoredBalancesFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (coloredBalances *ColoredBalances, err error) {
+	balancesCount, err := marshalUtil.ReadUint32()
+	if err != nil {
+		err = xerrors.Errorf("failed to parse element count (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+
+	var previousColor *Color
+	coloredBalances = NewColoredBalances(nil)
+	for i := uint32(0); i < balancesCount; i++ {
+		color, colorErr := ColorFromMarshalUtil(marshalUtil)
+		if colorErr != nil {
+			err = xerrors.Errorf("failed to parse Color from MarshalUtil: %w", colorErr)
+			return
+		}
+
+		// check semantic correctness (ensure ordering)
+		if previousColor != nil && bytes.Compare(previousColor[:], color[:]) != -1 {
+			err = xerrors.Errorf("parsed Colors are not in correct order: %w", ErrParseBytesFailed)
+			return
+		}
+
+		balance, balanceErr := marshalUtil.ReadUint64()
+		if balanceErr != nil {
+			err = xerrors.Errorf("failed to parse balance (%v): %w", balanceErr, ErrParseBytesFailed)
```

Comment:
Make this error more precise.
```suggestion
			err = xerrors.Errorf("failed to parse balance of Color %s (%v): %w", color.String(), balanceErr, ErrParseBytesFailed)
```

Reply:
Our errors already provide stack traces so I prefer to not be "too specific" about errors as this information is already provided by the stack trace and adding the "context" kind of becomes ambigious but I guess its up discussion.